client-coroutines/src/main/kotlin/io/provenance/client/coroutines/PbCoroutinesClient.kt

```diff
@@ -122,31 +129,67 @@ open class PbCoroutinesClient(
     suspend fun broadcastTx(
         baseReq: BaseReq,
         gasEstimate: GasEstimate,
-        mode: ServiceOuterClass.BroadcastMode = ServiceOuterClass.BroadcastMode.BROADCAST_MODE_SYNC
-    ): ServiceOuterClass.BroadcastTxResponse {
-        return cosmosService.broadcastTx(
-            ServiceOuterClass.BroadcastTxRequest
-                .newBuilder()
-                .setTxBytes(buildTx(baseReq, gasEstimate).toByteString())
-                .setMode(mode)
-                .build()
-        )
+        mode: BroadcastMode = BroadcastMode.BROADCAST_MODE_SYNC
+    ): BroadcastTxResponse {
+        return buildTx(baseReq, gasEstimate)
+            .emulateBlockMode(mode, baseReq.body.timeoutHeight) {
+               cosmosService.broadcastTx(it)
+            }
     }
 
     suspend fun estimateAndBroadcastTx(
         txBody: TxBody,
         signers: List<BaseReqSigner>,
-        mode: ServiceOuterClass.BroadcastMode = ServiceOuterClass.BroadcastMode.BROADCAST_MODE_SYNC,
+        mode: BroadcastMode = BroadcastMode.BROADCAST_MODE_SYNC,
         gasAdjustment: Double? = null,
         feeGranter: String? = null,
         feePayer: String? = null,
-    ): ServiceOuterClass.BroadcastTxResponse = baseRequest(
+    ): BroadcastTxResponse = baseRequest(
         txBody = txBody,
         signers = signers,
         gasAdjustment = gasAdjustment,
         feeGranter = feeGranter,
         feePayer = feePayer,
     ).let { baseReq -> broadcastTx(baseReq, estimateTx(baseReq), mode) }
+
+    private suspend fun TxOuterClass.TxRaw.emulateBlockMode(
+        mode: BroadcastMode,
+        providedTimeoutHeight: Long,
+        handler: suspend (BroadcastTxRequest) -> BroadcastTxResponse
+    ): BroadcastTxResponse {
+        val (actualMode, simulateBlock) = if (mode == BroadcastMode.BROADCAST_MODE_BLOCK) {
+            BroadcastMode.BROADCAST_MODE_SYNC to true
+        } else {
+            mode to false
+        }
+        return handler(BroadcastTxRequest.newBuilder()
+            .setTxBytes(this.toByteString())
+            .setMode(actualMode)
+            .build()
+        ).let { res ->
+            if (simulateBlock) {
+                val timeoutHeight = providedTimeoutHeight.takeIf { it > 0 } ?: (latestHeight() + 10) // default to 10 block timeout for polling if no height set
+                val txHash = res.txResponse.txhash
+                do {
+                    try {
+                        val tx = cosmosService.getTx(getTxRequest { hash = txHash })
+                        return res.toBuilder()
+                            .setTxResponse(tx.txResponse)
+                            .build()
+                    } catch (e: StatusException) {
```

Comment:
Apparently the coroutine grpc stuff throws a StatusException instead of a StatusRuntimeException? Anyone know why that is? Do you think I should add a second exception to extend from this instead for the coroutine implementation of this whole thing?